compiler/rustc_middle/src/middle/stability.rs

```diff
@@ -381,7 +359,7 @@ impl<'tcx> TyCtxt<'tcx> {
                 // With #![staged_api], we want to emit down the whole
                 // hierarchy.
                 let depr_attr = &depr_entry.attr;
-                if !skip || depr_attr.is_since_rustc_version {
+                if !skip || matches!(depr_attr.since, Some(DeprecatedSince::RustcVersion(_))) {
```

Comment:
It's okay either way, I think. Here is what having a method looks like: 8b8906b2641ab2c7b852ac956b7388cb614e5391.

It's not that widely useful because most places that look for `DeprecatedSince::RustcVersion(version)` also want to do something with the `version`. I found 2 places that didn't (the ones in that commit link). We could do `fn get_since_rustc_version(&self) -> Option<RustcVersion>` which is more broadly applicable, and then instead of `.is_since_rustc_version()` do `.get_since_rustc_version().is_some()`, but at that point it's no longer better than pattern matching the usual ways.